xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/internal/macro/DefaultMacroCategoryManager.java

```diff
@@ -161,4 +186,9 @@ private void addToCategory(Map<String, Set<MacroId>> result, MacroId macroId, St
         ids.add(macroId);
         result.put(macroCategory, ids);
     }
+
+    private Set<String> splitCategories(String s)
+    {
+        return Arrays.stream(s.split("\\s*,\\s*")).collect(Collectors.toSet());
```

Comment:
If you really want/need to use a regular expression here, this expression should be pre-compiled as a static variable (this should also produce a warning in SonarLint). Further, why aren't you using our standard list parsing implementation that we normally use for configuration variables?

Also, I think this discards the order of the categories (or at least doesn't guarantee it), is this intentional?

Reply:
Sonar is not raising an error (at least not in my ide), but indeed per-compilling the regex is a good idea.
`MacroTransformationConfiguration#getCategories` returns a `java.util.Properties` and not a `commons-configuration` Properties object. And it does not have the notion of list.